packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/ChartAnnotation.java

```diff
@@ -0,0 +1,54 @@
+package software.uncharted.terarium.hmiserver.models.dataservice;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.hypersistence.utils.hibernate.type.json.JsonType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import java.util.UUID;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+import org.hibernate.annotations.Type;
+import software.uncharted.terarium.hmiserver.annotations.TSModel;
+import software.uncharted.terarium.hmiserver.models.TerariumAsset;
+
+@Data
+@Accessors(chain = true)
+@EqualsAndHashCode(callSuper = true)
+@TSModel
+@Entity
+public class ChartAnnotation extends TerariumAsset {
+
+	@TSModel
+	public enum ChartType {
+		@JsonAlias("variable")
+		VARIABLE,
+		@JsonAlias("parameter")
+		PARAMETER,
+		@JsonAlias("loss")
+		LOSS,
+		@JsonAlias("error")
+		ERROR
+	}
```

Comment:
I am wondering if we want to hold off adding the enum and leave it as a TODO, as I am not sure we have the full listing of chart-types, and since enums are translated into constraints via hibernate it makes things messy if we want to add or remove types.

Otherwise we'd need to be as comprehensive as possible with all possible charts upfront.

Reply:
Yeah we probably are okay without the  `chartType` at the moment considering we are creating an annotation for each individual chart, I can comment out for now if updating enum value might be messy. But I'm also curious on how the migration happens with enum values with hibernate. Does it automatically handles the migration when adding new enum values or we need a separate migration script? What about delete or update cases? @dvince2

Reply:
For just _adding_ cases everything should be handled by `magic` and you're good.  However any time you want to remove, rename or reorder the enum then you'll be writing a migration for that

Reply:
I left tis as TODO